source/js/main.js

```diff
@@ -293,15 +293,90 @@ function initializeCreateRecipeButtons() {
   const createButton = document.querySelector('.create-recipe-button');
   createButton.addEventListener('click', createRecipeClicked);
 }
+
+function removeAllChildNodes(parent) {
+  while (parent.firstChild) {
+      parent.removeChild(parent.firstChild);
+  }
+}
+
+/**
+ * active when search button click
+ * store Recipe to local with query as category name
+ * Does NOT apply any filter & sort, only search by name
+ * 1.add even handler to search button to fetch recipes for query 
+ * 2.add fetched recipes to local Storage
+ * 3.populate recipe cards for search result page
+ * 4.TODO: if no enough result, call getRecipesByAutocomplete
+ */
+ async function searchButtonHandler() {
```

Comment:
I defined a function `openSearchResults` above already. please move all of the code for this function inside that one and leave the router.navigate as the last line there.

Reply:
openSearchResults is not an async function, may I know how to use the async API fetcher inside this function?

Reply:
you can make it async just change the signature to 
```
const openSearchResults = async () => {
}
```